src/site/topic20.rst

```diff
@@ -27,6 +27,31 @@ Iterative
 Recursive
 ---------
 
+* Now that we are more familiar with recursion, here is the exact same high-level abstract idea, but implemented with recursion
+
+.. code-block:: java
+    :linenos:
+
+    public static <T> int recursiveLinearSearch(T toFind, T[] data, int currentIndex) {
+        // Not Found
+        if (currentIndex == data.length) {
+            return -1;
+        } else if (data[currentIndex].equals(toFind)) {
+            return currentIndex;
+        } else {
+            return recursiveLinearSearch(toFind, data, currentIndex + 1);
+        }
+    }
+
+* If I wanted to call this method, I would start with ``currentIndex`` as ``0``
+    * ``recursiveLinearSearch(someTarget, someArray, 0)``
```

Comment:
It's typical to have a `public` and `private` method for these cases. Something like:
```java
public int indexOf(int x, int[] data) { return indexOf(x, data, 0); }
private int indexOf(int x, int[] data, int index) { /* search... */ }
```
May be worth mentioning in passing -- recursive methods don't always have a weirdo public method signature.